src/modules/facility/dto/create-facility-request.dto.ts

```diff
@@ -0,0 +1,190 @@
+import { EXAMPLES, PROPERTIES } from '@ukef/constants';
+import { ValidatedCurrencyApiProperty } from '@ukef/decorators/validated-currency-api-property-decorator';
+import { ValidatedDateOnlyApiProperty } from '@ukef/decorators/validated-date-only-api-property.decorator';
+import { ValidatedDealIdentifierApiProperty } from '@ukef/decorators/validated-deal-identifier-api-property.decorator';
+import { ValidatedFacilityIdentifierApiProperty } from '@ukef/decorators/validated-facility-identifier-api-property';
+import { ValidatedNumberApiProperty } from '@ukef/decorators/validated-number-api-property.decorator';
+import { ValidatedPartyIdentifierApiProperty } from '@ukef/decorators/validated-party-identifier-api-property.decorator';
+import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
+import { DateOnlyString } from '@ukef/helpers';
+
+export type CreateFacilityRequest = CreateFacilityRequestItem[];
+
+// TODO APIM-69: description improvements if possible
+export class CreateFacilityRequestItem {
+  @ValidatedDealIdentifierApiProperty({
+    description: 'The identifier of the deal for the facility.',
+  })
+  dealIdentifier: string;
```

Comment:
do we need these properties to be `readOnly`?

Reply:
I have added `readonly` to them